### DISCUSSION TITLE
Open web links that exists in pdf in the same modal and not in the page

### DISCUSSION THREAD

**Discussion by mkhadiri1** [stephanrauh/ngx-extended-pdf-viewer#1534] [General]
Can we force open the web links that exist in the pdf in the modal and not in the page pls?

**Comment by stephanrauh** [comment]
Yes, you can, with some effort. I guess you need to catch one of the events `(textLayerRendered)` or `(pageRendered)`. Strictly speaking, `(annotationLayerEvent)` is an even better choice - but I didn't implement this event yet. Nonetheless, I hope the other event don't fire too early. If they do, add a short timeout.

The event listener needs to find the DOM of the annotation layer of the page that's just been rendered. If that sounds confusing - don't be. Inspect the HTML tree in your browsers developer tools, and you'll quickly see what I'm talking about. Next, add a `click` listener to each link in the annotation layer.

The final challenge is to make Angular aware of your click. I've documented how to do this here: https://pdfviewer.net/extended-pdf-viewer/custom-thumbnails (check out the two tabs called "Call Angular code"). The example is about thumbnails, but it's the same problem: how to call Angular code from a pure JavaScript `click` listener.
